File: app/src/main/java/com/lkms/data/model/java/InventoryTransaction.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "InventoryTransaction".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class InventoryTransaction {

    @SerializedName("transactionId")
    private Integer transactionId;

    @SerializedName("transactionType")
    private String transactionType;

    // Khóa ngoại đến Item
    @SerializedName("itemId")
    private Integer itemId;

    // Khóa ngoại đến User
    @SerializedName("userId")
    private Integer userId;

    @SerializedName("quantity")
    private Integer quantity;

    @SerializedName("transactionTime")
    private String transactionTime;

    @SerializedName("transactionStatus")
    private String transactionStatus;

    @SerializedName("rejectReason")
    private String rejectReason;
}

File: app/src/main/java/com/lkms/data/model/InventoryTransaction.kt
// Trong file: data/model/InventoryTransaction.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "InventoryTransaction".
 */
@Serializable
data class InventoryTransaction(
    @SerialName("transactionId")
    val transactionId: Int,

    @SerialName("transactionType")
    val transactionType: String? = null,

    // Khóa ngoại đến Item
    @SerialName("itemId")
    val itemId: Int? = null,

    // Khóa ngoại đến User
    @SerialName("userId")
    val userId: Int? = null,

    @SerialName("quantity")
    val quantity: Int? = null,

    @SerialName("transactionTime")
    val transactionTime: String? = null,

    @SerialName("transactionStatus")
    val transactionStatus: String? = null,

    @SerialName("rejectReason")
    val rejectReason: String? = null
)